src/EventStore.Common/Configuration/ConfigurationBuilderExtensions.cs

```diff
@@ -28,5 +33,59 @@ public static class ConfigurationBuilderExtensions {
 
 			return builder;
 		}
+
+		// Allows configuration to be mounted inside a specified section
+		public static IConfigurationBuilder AddSection(
+			this IConfigurationBuilder self,
+			string sectionName,
+			Action<IConfigurationBuilder> configure) =>
+			
+			self.Add(new SectionSource(sectionName, configure));
+
+		public static IConfigurationBuilder AddConfigFile(
+			this IConfigurationBuilder builder,
+			string configFilePath,
+			bool optional = false,
+			bool reloadOnChange = false) {
+
+			if (!Locations.TryLocateConfigFile(configFilePath, out var directory, out var fileName)) {
+				if (optional) {
+					return builder;
+				}
+
+				throw new FileNotFoundException(
+					$"Could not find {configFilePath} in the following directories: {string.Join(", ", Locations.GetPotentialConfigurationDirectories())}");
+			}
+
+			builder.AddJsonFile(config => {
```

Comment:
Could we instead pass just the `filename`, `optional` and `reloadOnChange` parameters?

Reply:
im not sure, i've kept it the same way as it was being called before